### PR TITLE
Update BCL versions to .NET 7

### PR DESCRIPTION
**Pull request by genlu**
VS and Msbuild already did  this
https://github.com/dotnet/msbuild/pull/8347/files#

[validation build](https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/451139)

### CODE REVIEW

**Review by jaredpar — APPROVED** [review]

File: eng/Versions.props
Comment (333fred) [inline eng/Versions.props:235]: Is it worth adding a property for these versions?

**Comment by genlu** [comment]
> error : (NETCORE_ENGINEERING_TELEMETRY=Build) System.Collections.Immutable 7.0.0 doesn't support netcoreapp3.1 and has not been tested with it.

@jaredpar now might be a good time to change our tfm to .NET 6

**Comment by jaredpar** [comment]
> @jaredpar now might be a good time to change our tfm to .NET 6

Yeah. Let me check with @agocke as he had a PR that did this and already fixed up the inevitable conflicts we had.

**Comment by agocke** [comment]
https://github.com/dotnet/roslyn/pull/66519 does that (mostly) but it's limited to MS.CA, MS.CA#, and MS.CA.VB. It's not bad to start with that, though. I can split that change out.

**Comment by genlu** [comment]
@agocke OK. So the plan is, you are gonna add net6 tfm to compiler projects. Once it's merged, I can change IDE projects to targeting net6 and remove netcore31 from compilers?

**Comment by agocke** [comment]
Works for me, let me split out that commit.

**Comment by jaredpar** [comment]
> So the plan is, you are gonna add net6 tfm to compiler projects.

Not add, replace `netcoreapp3.1` with `net6`.

**Comment by agocke** [comment]
Pr for that is at https://github.com/dotnet/roslyn/pull/66565. To be honest, I'm stumped on the correctness failures. Building locally doesn't produce the same problems, and the error is a confusing one from the NuGet tooling.

**Comment by jaredpar** [comment]
> To be honest, I'm stumped on the correctness failures. 

Didn't I teach you anything? Never ask why ... cause know I'm going to tell you by putting you on the internal threads. 😉 

TLDR: requeue your build

**Comment by genlu** [comment]
> > So the plan is, you are gonna add net6 tfm to compiler projects.
> 
> Not add, replace `netcoreapp3.1` with `net6`.

Well, then you'd need to retarget all IDE projects in the PR tpp, wouldn't you?

**Comment by agocke** [comment]
Nope, builds work anyway. It's possible some stuff is now using the netstandard2.0 versions, but that doesn't seem like a big deal as a point-in-time issue.

**Comment by genlu** [comment]
Blocked on integration tests. We can't merge this until VS 17.6 preview is released

**Review by arkalyanms — APPROVED** [review]

**Pull request merged**
